Solutions/Branwen/GUI.cs

```diff
@@ -9,324 +9,460 @@
+
+					//PartOfPath
+					insertPathsStatement += i + ", ";
+
+					//Path
+					insertPathsStatement += "'" + file.Path[i].Replace("'", "''") + "', ";
+
+					//MediaDrive
+					insertPathsStatement += file.MediaDrive + ")";
+				}
+
+				fileCount++;
+			}
+			insertFilesStatement += ";";
+			insertPathsStatement += ";";
+
+			MySqlCommand insertCommand = new MySqlCommand(insertFilesStatement, mySqlConnection);
+			insertCommand.ExecuteNonQuery();
+			insertCommand = new MySqlCommand(insertPathsStatement, mySqlConnection);
+			insertCommand.ExecuteNonQuery();
+		}
+
+		/// <summary>
+		/// Writes an Excel document with all the contentes of the database
+		/// </summary>
+		/// <param name="mySqlConnection"></param>
+		/// <param name="outputFile"></param>
+		private void ExportDatabaseToFile(MySqlConnection mySqlConnection, string outputFile)
+		{
+			//Get all the MediaTypes
+			List<string> mediaTypes = new List<string>();
+			MySqlCommand command = new MySqlCommand("SELECT MediaType FROM SaurutobiMediaFiles GROUP BY MediaType;", mySqlConnection);
+			using (MySqlDataReader mediaTypeReader = command.ExecuteReader())
+			{
+				while (mediaTypeReader.Read())
+				{
+					mediaTypes.Add(mediaTypeReader["MediaType"].ToString());
+				}
+				mediaTypeReader.Close();
+			}
+
+			//Make sure the outputFile doesn't exist
+			if (File.Exists(outputFile))
+			{
+				File.Delete(outputFile);
+			}
+
+			//Create worksheet
+			SpreadsheetDocument spreadsheetDocument = SpreadsheetDocument.Create(outputFile, DocumentFormat.OpenXml.SpreadsheetDocumentType.Workbook);
+			WorkbookPart workbookpart = spreadsheetDocument.AddWorkbookPart();
+			workbookpart.Workbook = new Workbook();
+			Sheets sheets = spreadsheetDocument.WorkbookPart.Workbook.AppendChild(new Sheets());
+
+			//Loop through all MediaTypes and make a new sheet and put that shit in there
+			for (int i = 0; i < mediaTypes.Count; i++)
+			{
+				WorksheetPart worksheetPart = workbookpart.AddNewPart<WorksheetPart>();
+				worksheetPart.Worksheet = new Worksheet(new SheetData());
+				SheetData sheetData = worksheetPart.Worksheet.GetFirstChild<SheetData>();
+				Sheet sheet = new Sheet();
+				sheet.Id = spreadsheetDocument.WorkbookPart.GetIdOfPart(worksheetPart);
+				sheet.Name = mediaTypes[i];
+				sheet.SheetId = (UInt32)Convert.ToInt32(i + 1);
+				sheets.Append(sheet);
+
+				List<BranwenFileInfo> files = new List<BranwenFileInfo>();
+
+				command = new MySqlCommand("SELECT FileName, Extension, Size, MediaDrive FROM SaurutobiMediaFiles WHERE MediaType = '" + mediaTypes[i] + "';", mySqlConnection);
+				using (MySqlDataReader fileReader = command.ExecuteReader())
+				{
+					while (fileReader.Read())
+					{
+						BranwenFileInfo newFile = new BranwenFileInfo();
+						newFile.Name = fileReader["FileName"].ToString();
+						newFile.Extension = fileReader["Extension"].ToString();
+						newFile.FileSize = long.Parse(fileReader["Size"].ToString());

```

Comment:
yup, shoulda done that. Lazy programming

On Thu, Dec 17, 2015 at 11:25 PM, Alex Ayerdi <notifications@github.com>
wrote:

> In Solutions/Branwen/GUI.cs
> <https://github.com/LbKStudios/Branwen/pull/5#discussion_r48000366>:
>
> > +				sheet.Id = spreadsheetDocument.WorkbookPart.GetIdOfPart(worksheetPart);
> > +				sheet.Name = mediaTypes[i];
> > +				sheet.SheetId = (UInt32)Convert.ToInt32(i + 1);
> > +				sheets.Append(sheet);
> > +
> > +				List<BranwenFileInfo> files = new List<BranwenFileInfo>();
> > +
> > +				command = new MySqlCommand("SELECT FileName, Extension, Size, MediaDrive FROM SaurutobiMediaFiles WHERE MediaType = '" + mediaTypes[i] + "';", mySqlConnection);
> > +				using (MySqlDataReader fileReader = command.ExecuteReader())
> > +				{
> > +					while (fileReader.Read())
> > +					{
> > +						BranwenFileInfo newFile = new BranwenFileInfo();
> > +						newFile.Name = fileReader["FileName"].ToString();
> > +						newFile.Extension = fileReader["Extension"].ToString();
> > +						newFile.FileSize = long.Parse(fileReader["Size"].ToString());
>
> TryParse perhaps for safer conversion
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/LbKStudios/Branwen/pull/5/files#r48000366>.
>